### PR TITLE
Nitrile and latex gloves reduce firefighter carry delay 

### PR DESCRIPTION
**Pull request by Cenrus**
## About The Pull Request

Wearing nitrile or latex gloves reduces the time it takes to fireman carry a player.

it takes ages to lift people up, and in an emergency that can really heck your day. now it's faster with latex or nitrile gloves. The latex gloves sprite is bugged and won't show on your hands, I would appreciate someone looking into this.

Mirrored from: https://github.com/tgstation/tgstation/pull/46735

## Changelog

:cl:
add: Halved firefight carry delay with latex or nitrile gloves
/:cl:

### CODE REVIEW

**Comment by jlsnow301** [comment]
Might convince people to actually carry them rather than bleed them across the entire hallway

File: code/modules/clothing/gloves/color.dm [hunk 1]
```diff
@@ -158,23 +158,34 @@
...
+	var/carrytrait = TRAIT_QUICK_CARRY
 
+/obj/item/clothing/gloves/color/latex/equipped(mob/user, slot)
+	..()
+	if(slot == SLOT_GLOVES)
+		ADD_TRAIT(user, carrytrait, CLOTHING_TRAIT)
+
+/obj/item/clothing/gloves/color/latex/dropped(mob/user)
+	..()
+	REMOVE_TRAIT(user, carrytrait, CLOTHING_TRAIT)
```
Comment (ike709) [inline code/modules/clothing/gloves/color.dm:178]: You need to make a `Destroy()` subtype as well with similar functionality.

File: code/modules/clothing/gloves/color.dm [hunk 2]
```diff
@@ -158,23 +158,34 @@
...
+	
 /obj/item/clothing/gloves/color/latex/nitrile
 	name = "nitrile gloves"
-	desc = "Pricy sterile gloves that are stronger than latex."
+	desc = "Pricy sterile gloves that are thicker than latex. Transfers intimate paramedic knowledge into the user via nanochips."
```
Comment (ike709) [inline code/modules/clothing/gloves/color.dm:182]: As someone who regularly works with both types of gloves, `stronger` would technically be more correct. Nitrile gloves are still paper thin, they're just a sturdier material.

File: code/modules/mob/living/carbon/human/human.dm [hunk 3]
```diff
@@ -886,10 +886,20 @@
 	return (ishuman(target) && !(target.mobility_flags & MOBILITY_STAND))
 
 /mob/living/carbon/human/proc/fireman_carry(mob/living/carbon/target)
-	if(can_be_firemanned(target))
-		visible_message("<span class='notice'>[src] starts lifting [target] onto their back...</span>",
-			"<span class='notice'>You start lifting [target] onto your back...</span>")
-		if(do_after(src, 50, TRUE, target))
+	var/carrydelay = 50 //if you have latex you are faster at grabbing
+	var/skills_space = "" //cobby told me to do this
```
Comment (ike709) [inline code/modules/mob/living/carbon/human/human.dm:890]: > cobby

Is this PR an uncredited TG port?

File: code/modules/mob/living/carbon/human/human.dm [hunk 4]
```diff
@@ -886,10 +886,20 @@
...
+	if(HAS_TRAIT(src, TRAIT_QUICKER_CARRY))
+		carrydelay = 30
+		skills_space = "expertly"
+	else if(HAS_TRAIT(src, TRAIT_QUICK_CARRY))
+		carrydelay = 40
+		skills_space = "quickly"
+	if(can_be_firemanned(target) && !incapacitated(FALSE, TRUE))
+		visible_message("<span class='notice'>[src] starts [skills_space] lifting [target] onto their back..</span>",
+		//Joe Medic starts quickly/expertly lifting Grey Tider onto their back..
+		"<span class='notice'>[carrydelay < 35 ? "Using your gloves' nanochips, you" : "You"] [skills_space] start to lift [target] onto your back[carrydelay == 40 ? ", while assisted by the nanochips in your gloves.." : "..."]</span>")
```
Comment (ike709) [inline code/modules/mob/living/carbon/human/human.dm:900]:
```suggestion
		"<span class='notice'>[HAS_TRAIT(src, TRAIT_QUICKER_CARRY) ? "Using your gloves' nanochips, you" : "You"] [skills_space] start to lift [target] onto your back[HAS_TRAIT(src, TRAIT_QUICK_CARRY) ? ", while assisted by the nanochips in your gloves.." : "..."]</span>")
```

File: code/modules/clothing/gloves/color.dm [hunk 5]
```diff
@@ -158,23 +158,34 @@
 
 /obj/item/clothing/gloves/color/latex
 	name = "latex gloves"
-	desc = "Cheap sterile gloves made from latex."
+	desc = "Cheap sterile gloves made from latex. Transfers minor paramedic knowledge to the user via budget nanochips."
 	icon_state = "latex"
-	item_state = "lgloves"
```
Comment (ike709) [inline code/modules/clothing/gloves/color.dm:163]: What's with the random icon change?

**Review by ike709 — CHANGES_REQUESTED** [review]
Address the things.

**Comment by ike709** [comment]
Stale.

**Pull request closed without merging**